mmv1/third_party/terraform/tests/resource_dataform_repository_test.go

```diff
@@ -0,0 +1,127 @@
+		ProtoV5ProviderFactories: ProtoV5ProviderBetaFactories(t),
+		CheckDestroy:             testAccCheckDataformRepositoryDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataformRepository_basic(context),
+			},
+			{
+				ResourceName:            "google_dataform_repository.dataform_respository",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"region"},
+			},
+			{
+				Config: testAccDataformRepository_updated(context),
+			},
+			{
+				ResourceName:            "google_dataform_repository.dataform_respository",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"region"},
+			},
+		},
+	})
+}
+
+func testAccDataformRepository_basic(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_sourcerepo_repository" "git_repository" {
+  provider = google-beta
+  name = "my/repository%{random_suffix}"
+}
+
+resource "google_secret_manager_secret" "secret" {
+  provider = google-beta
+  secret_id = "secret"
+
+  replication {
+    automatic = true
+  }
+}
+
+resource "google_secret_manager_secret_version" "secret_version" {
+  provider = google-beta
+  secret = google_secret_manager_secret.secret.id
+
+  secret_data = "tf-test-secret-data%{random_suffix}"
+}
+
+resource "google_dataform_repository" "dataform_respository" {
+  provider = google-beta
+  name = "tf_test_dataform_repository%{random_suffix}"
+
+  git_remote_settings {
+      url = google_sourcerepo_repository.git_repository.url
+      default_branch = "main"
+      authentication_token_secret_version = google_secret_manager_secret_version.secret_version.id
+  }
+
+  workspace_compilation_overrides {
+    default_database = "database"
+    schema_suffix = "_suffix"
+    table_prefix = "prefix_"
+  }
+}
+`, context)
+}
+
+func testAccDataformRepository_updated(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_sourcerepo_repository" "git_repository" {
+  provider = google-beta
+  name = "my/repository%{random_suffix}"
+}
+
+resource "google_secret_manager_secret" "secret" {
+  provider = google-beta
+  secret_id = "secret"
+
+  replication {
+    automatic = true
+  }
+}
+
+resource "google_secret_manager_secret_version" "secret_version" {
+  provider = google-beta
+  secret = google_secret_manager_secret.secret.id
+
+  secret_data = "tf-test-secret-data%{random_suffix}"
+}
+
+resource "google_dataform_repository" "dataform_respository" {
+  provider = google-beta
+  name = "tf_test_dataform_repository%{random_suffix}"
+
+  git_remote_settings {
+      url = google_sourcerepo_repository.git_repository.url
+      default_branch = "main"
+      authentication_token_secret_version = google_secret_manager_secret_version.secret_version.id
+  }
+
+  workspace_compilation_overrides {
+    default_database = "database_v2"
+    schema_suffix = "_suffix_v2"
```

Comment:
```suggestion
```

Can we test removing one of these values?

Reply:
I deleted the field: `default_database` in [ca92493](https://github.com/GoogleCloudPlatform/magic-modules/pull/8074/commits/ca92493b72c54ed4145060258b1acf8fa20695bb).

Reply:
In addition to that, I updated the imported module to fix the error on terraform-provider-google-build-and-unit-tests with 5417e7f.

Reply:
@rileykarson google_dataform_respository is a beta resource. Therefore, testAccCheckDataformRepositoryDestroyProducer was not generated along with an auto-generated test in ` terraform-provider-google-build-and-unit-tests`

```
Error: google/resource_dataform_repository_test.go:23:2[9](https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/5158311281/jobs/9291839419#step:8:10): undefined: testAccCheckDataformRepositoryDestroyProducer
```
from https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/5158311281/jobs/9291839419

The unit tests passed locally under terraform-provider-google-beta. I'll copy the function with a different name as mitigation.

Reply:
Oh, we'll need to guard the test file: https://googlecloudplatform.github.io/magic-modules/how-to/add-handwritten-test/#testing-beta-features

Reply:
@rileykarson Thank you for your feedback! I added the guard in [aaa9bc1](https://github.com/GoogleCloudPlatform/magic-modules/pull/8074/commits/aaa9bc1867e537300d72a81de442602e266a37a5). I added `autogen_exception` too along with the other handwritten tests. Please let me know if you have a concern. Thanks